### PR TITLE
V1 reusable rate limiter

### PR DESCRIPTION
**Pull request by jalvz**

### CODE REVIEW

File: beater/common_handlers.go [hunk 1]
```diff
@@ -304,19 +302,15 @@ func killSwitchHandler(killSwitch bool, h http.Handler) http.Handler {
...
 
 const (
-	rateLimitCacheSize       = 1000
-	rateLimitBurstMultiplier = 2
+	rateLimitCacheSize = 1000
 )
 
 func ipRateLimitHandler(rateLimit int, h http.Handler) http.Handler {
-	cache, _ := lru.New(rateLimitCacheSize)
+	rlc, _ := NewRlCache(rateLimitCacheSize, rateLimit, rateLimitBurstMultiplier)
```
Comment (simitt) [inline beater/common_handlers.go]: Why did you remove the `rateLimitBurstMultiplier = 2` const and where does `rateLimitBurstMultiplier` now come from?

File: beater/route_config.go [hunk 2]
```diff
@@ -228,7 +228,7 @@ func (v v2Route) Handler(url string, c *Config, report publish.Reporter) http.Ha
 	}
 
 	if url == V2RumURL {
-		if rlc, err := NewRlCache(c.RumConfig.EventRate.LruSize, c.RumConfig.EventRate.Limit); err == nil {
+		if rlc, err := NewRlCache(c.RumConfig.EventRate.LruSize, c.RumConfig.EventRate.Limit, burstMultiplier); err == nil {
```
Comment (simitt) [inline beater/route_config.go]: I think you should now remove https://github.com/elastic/apm-server/pull/1459/files#diff-48b2aa9b71bf27a37b9122608aa3dfb1R28, and instead create a `v2BurstMultiplier` const here.

File: beater/common_handlers.go (hunk 1 above)
Comment (jalvz) [reply beater/common_handlers.go]: sorry, that was a bad commit - fixed now
